controller/api/destination/watcher/opaque_ports_watcher.go

```diff
@@ -35,12 +38,21 @@ type (
 	}
 )
 
+var opaquePortsMetrics = promauto.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "service_subscribers",
+		Help: "Number of subscribers to Service changes.",
+	},
+	[]string{"namespace", "name"},
+)
```

Comment:
calling `promauto.NewGaugeVec()` and friends at the package level can cause "already registered" errors, particularly when writing tests.

recommend one of the following:
1) call `prometheus.NewGaugeVec()` here, and then `prometheus.MustRegister()` in `NewOpaquePortsWatcher()` (or in other cases, if you have a `Start()` method, it's even better to register there them, as you're less likely to call a `Start()` method in a test).
2) move `promauto.NewGaugeVec()` into `NewOpaquePortsWatcher()`.

more detail here:
https://github.com/prometheus/client_golang/blob/2fced9611046ce27d6088eb8930541d1534138a4/prometheus/promauto/auto.go#L126-L159

Reply:
Turns out `NewOpaquePortsWatcher()` is called from a unit test, repeatedly. So that's why I declared and registered the metric at the package level, so it only gets called once. Does that make sense, or am I missing something?

Reply:
Ah makes sense! I think in this case it's fine to merge as-is. Ideally, I think it's best practice to avoid registering metrics when possible, for the reasons laid out in `client_golang`, but again for this use-case it works. 👍